core/src/calleeFunctions/helpers/oneInch.ts

```diff
@@ -0,0 +1,161 @@
+    const protocolIds = response.protocols.map(protocol => protocol.id);
+    return protocolIds.filter(protocolId => !protocolId.toLowerCase().includes('limit'));
+}
+
+export async function getOneinchSwapParameters(
+    network: string,
+    collateralSymbol: string,
+    amount: string,
+    marketId: string,
+    slippage = '10'
+): Promise<OneInchSwapRepsonse> {
+    const isFork = getNetworkConfigByType(network).isFork;
+    const chainId = isFork ? 1 : parseInt(getChainIdByNetworkType(network) || '', 16);
+    if (!isFork && !SUPPORTED_1INCH_NETWORK_IDS.includes(chainId)) {
+        throw new Error(`1inch does not support network ${network}`);
+    }
+    if (Number.isNaN(chainId)) {
+        throw new Error(`Invalid chainId: ${chainId}`);
+    }
+    const toTokenAddress = await getTokenAddressByNetworkAndSymbol(network, 'DAI');
+    const fromTokenAddress = await getTokenAddressByNetworkAndSymbol(network, collateralSymbol);
+    const calleeAddress = getCalleeAddressByCollateralType(
+        network,
+        getCollateralConfigBySymbol(collateralSymbol).ilk,
+        marketId
+    );
+    // Documentation https://docs.1inch.io/docs/aggregation-protocol/api/swap-params/
+    const swapParams = {
+        fromTokenAddress,
+        toTokenAddress,
+        fromAddress: calleeAddress,
+        amount,
+        slippage,
+        allowPartialFill: false, // disable partial fill
+        disableEstimate: true, // disable eth_estimateGas
+        compatibilityMode: true, // always receive parameters for the `swap` call
+    };
+    const oneinchResponse = await executeOneInchApiRequest(chainId, '/swap', swapParams);
+    const functionSignature = ethers.utils.hexDataSlice(oneinchResponse.tx.data, 0, 4); // see https://docs.soliditylang.org/en/develop/abi-spec.html#function-selector
+    if (functionSignature !== EXPECTED_SIGNATURE) {
+        throw new Error(`Unexpected 1inch function signature: ${functionSignature}, expected: ${EXPECTED_SIGNATURE}`);
+    }
+    return oneinchResponse;
+}
+
+export async function extractPathFromSwapResponseProtocols(
+    network: string,
+    oneInchRoutes: OneInchSwapRoute[]
+): Promise<string[]> {
+    const pathStepsResolves = await Promise.all(
+        oneInchRoutes[0].map(async route => {
+            return await Promise.all([
+                await getErc20SymbolByAddress(network, route[0].fromTokenAddress),
+                await getErc20SymbolByAddress(network, route[0].toTokenAddress),
+            ]);
+        })
+    );
+    const path = [pathStepsResolves[0][0]];
+    for (const step of pathStepsResolves) {
+        path.push(step[1]);
+    }
+    return path;
+}
+
+export async function getOneInchMarketData(
+    network: string,
+    collateral: CollateralConfig,
+    amount: BigNumber,
+    marketId: string
+) {
+    const swapData = await getOneinchSwapParameters(
+        network,
+        collateral.symbol,
+        amount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(0),
```

Comment:
Why WAD if you're passing the amount of collateral?

Reply:
i don't get the question.
see api https://docs.1inch.io/docs/aggregation-protocol/api/swap-params swap parameters explainer for `amount` field.

Reply:
The collateral amount that you `.shiftedBy` here can have different different number of decimals depending on the collateral. WAD is always 18, but if your collateral is USDC it should be 6. We keep the amount of decimals inside `collateralConfig`
```suggestion
        amount.shiftedBy(collateral.decimals).toFixed(0),
```